src/hooks/table/use-select.tsx

```diff
@@ -0,0 +1,65 @@
+import { useCallback, useState, ChangeEvent } from 'react'
+
+interface Item {
+  _id: string
+}
+
+interface SelectHook {
+  selected: string[]
+  isSelected: (id: string) => boolean
+  clearSelected: () => void
+  createSelectAllHandler: (
+    items: Item[]
+  ) => (e: ChangeEvent<HTMLInputElement>) => void
+  handleSelectClick: (id: string) => void
+}
+
+const useSelect = ({
+  initialSelect = []
+}: {
+  initialSelect?: string[]
+}): SelectHook => {
+  const [selected, setSelected] = useState<string[]>(initialSelect)
+
+  const clearSelected = useCallback(() => setSelected([]), [setSelected])
+
+  const handleSelectAllClick = (
+    e: ChangeEvent<HTMLInputElement>,
+    items: Item[]
+  ) => {
+    if (e.target.checked) {
+      const newSelected = items.map((item) => item._id)
+      return setSelected(newSelected)
```

Comment:
Maybe it would be better to move `return` statement to a new line, because it is a bit confusing whether we return something from `setSelected` or we just quitung function `handleSelectAllClick`

Reply:
No, i meant like following:

```jsx
setSelected(newSelected)
return
```